cheetah/accelerator/segment.py

```diff
@@ -511,10 +511,10 @@ def plot_twiss(self, beam: Beam, ax: Optional[Any] = None) -> None:
             outgoing = element.track(longitudinal_beams[-1])
```

Comment:
So now the `plot_twiss` only works for segment with shape (1,), right? Probably should add a warning or explicitly raise some error if calling with batched segment

Reply:
Yeah ... actually right now **all** plot functions only use the first beam in a batch. I'm not really a fan of cluttering things with warnings (just because I know what that will look like). So either we at least mention it in the docs, and maybe add a warning, but with an option to turn that warning off?

Reply:
Or we add an argument to all plot function where you give the index you want from the batch.

Reply:
Yes, we should at least mention it in the doc, and have a default behavior when passing a batched segment, because now this will just fail.

Reply:
> Or we add an argument to all plot function where you give the index you want from the batch.

Or... I just realised we don't have a convenient function to return one instance of the batched segment yet

Reply:
> > Or we add an argument to all plot function where you give the index you want from the batch.
> 
> Or... I just realised we don't have a convenient function to return one instance of the batched segment yet

Like an `unbroadcast`? 😄

Reply:
I'm also not sure how all of this would play with the broadcast rules Ryan proposed.

Reply:
> > > Or we add an argument to all plot function where you give the index you want from the batch.
> > 
> > 
> > Or... I just realised we don't have a convenient function to return one instance of the batched segment yet
> 
> Like an `unbroadcast`? 😄

Sure, something like a `__getitem__(index)`. The potential usecases would be 
1. Plotting 
2. Pass a batch of settings, get a batch of output beam, and retrieve the best setitng with `segment[torch.argmax(some_beam_parameter)]`

Reply:
Yeah ... we should have this. I worry we should wait until the broadcasting is in to do that though.